rebasebot/bot.py

```diff
@@ -318,7 +270,13 @@ def _prepare_rebase_branch(gitwd: git.Repo, source: GitHubBranch, dest: GitHubBr
     gitwd.git.checkout("-b", MERGE_TMP_BRANCH, f"source/{source.branch}")
 
     # Make sure we are at the tip of our branch.
-    gitwd.git.checkout(f"dest/{dest.branch}")
+    gitwd.git.checkout(f"dest/{dest.branch}", force=True)
+
+    # Reset any staged changes
+    gitwd.git.reset('--hard')
+
+    # Clean untracked files and directories
+    gitwd.git.clean('-fd')
```

Comment:
I've run into issues with lingering untracked files when reusing the rebase directory.

I've only kept gitwd.git.clean('-fd') but moved it into _init_working_dir. To avoid silencing any errors from badly behaving PRE_REBASE lifecyclehooks.